src/kafka/connection/connection_manager.cc

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#include "connection_manager.hh"
+
+namespace seastar {
+
+namespace kafka {
+
+future<> connection_manager::connect(const std::string& host, uint16_t port) {
```

Comment:
Maybe modify the method to return `future<lw_shared_ptr<kafka_connection>>`? That way we don't have to `connect(...).then([]{ return get_connection(...); })`